tests/unit/api/test_model.py

```diff
@@ -246,3 +247,95 @@ def test_create_model_bad_name(client, building_motif):
     assert results.status_code == 400
 
     assert len(building_motif.table_connection.get_all_db_models()) == 0
+
+
+def test_validate_model(client, building_motif):
+    # Set up
+    library_1 = Library.load(ontology_graph="tests/unit/fixtures/shapes/shape1.ttl")
+    assert library_1 is not None
+    library_2 = Library.load(directory="tests/unit/fixtures/templates")
+    assert library_2 is not None
+
+    BLDG = Namespace("urn:building/")
+    model = Model.create(name=BLDG)
+    model.add_triples((BLDG["vav1"], A, BRICK.VAV))
+
+    # Action
+    results = client.get(
+        f"/models/{model.id}/validate?library_ids={library_1.id}{library_2.id}",
+        headers={"Content-Type": "application/json"},
+    )
+
+    # Assert
+    assert results.status_code == 200
+
+    assert results.get_json().keys() == {"message", "reasons", "valid"}
+    assert isinstance(results.get_json()["message"], str)
+    assert isinstance(results.get_json()["reasons"], list)
+    assert not results.get_json()["valid"]
+
+    # Set up
+    model.add_triples((BLDG["vav1"], A, BRICK.VAV))
+    model.add_triples((BLDG["vav1"], BRICK.hasPoint, BLDG["temp_sensor"]))
+    model.add_triples((BLDG["temp_sensor"], A, BRICK.Temperature_Sensor))
+    model.add_triples((BLDG["vav1"], BRICK.hasPoint, BLDG["flow_sensor"]))
+    model.add_triples((BLDG["flow_sensor"], A, BRICK.Air_Flow_Sensor))
+
+    # Action
+    results = client.get(
+        f"/models/{model.id}/validate?library_ids={library_1.id}",
+        headers={"Content-Type": "application/json"},
+    )
+
+    # Assert
+    assert results.status_code == 200
+
+    assert results.get_json().keys() == {"message", "reasons", "valid"}
+    assert isinstance(results.get_json()["message"], str)
+    assert results.get_json()["valid"]
+    assert isinstance(results.get_json()["reasons"], list)
+
+
+def test_validate_model_bad_model_id(client, building_motif):
+    # Set up
+    library = Library.load(ontology_graph="tests/unit/fixtures/shapes/shape1.ttl")
+    assert library is not None
+
+    # Action
+    results = client.get(
+        f"/models/{-1}/validate?library_ids={library.id}",
+        headers={"Content-Type": "application/json"},
+    )
+
+    # Assert
+    assert results.status_code == 404
+
+
+def test_validate_model_no_libraries(client, building_motif):
+    # Set up
+    BLDG = Namespace("urn:building/")
+    model = Model.create(name=BLDG)
+
+    # Action
+    results = client.get(
+        f"/models/{model.id}/validate",  # no libraries arg
+        headers={"Content-Type": "application/xml"},
```

Comment:
why is the `Content-Type` `application/xml` for this test?

Reply:
an oversight, I'll fix it